### ISSUE TITLE
Solver

### ISSUE THREAD

**Issue by EstherCentrifuge** [centrifuge/apps#883]
we are  looking at integrating the solver in V1.

This is a script that runs a linear solver on the epoch state and calculates the optimal solution. We built this for Tinlake, and need to adapt it for Pools on Cent Chain (as going from DROP/TIN to multiple tranches changes the constraints of the linear problem).

I started a bit of work on this a while back, just added this to centrifuge.js here: https://github.com/centrifuge/apps/pull/882/files It’s not working yet, you can see that one of the test cases is failing; this is because the constraints don’t handle the min risk buffers yet (see https://github.com/centrifuge/apps/pull/882/files#diff-6161a38f5baec9bb48160224a5223e893515048f075833237434f1c7463c013aR59).

I would suggest that I pair program with you on this, to onboard you on how all of this works and try to fix it together. Maybe we can find some time either next week or the week after next. Before that, I’d suggest taking a look at the code and potentially also the model that underpins it (https://docs.google.com/spreadsheets/d/16hpWBzGFxlhsIFYJYl1Im9BsNLKVjvJj8VUvECxqduE/edit#gid=1642750340).

After the tests are working, we will then need to integrate this with the` centrifuge.pools.submitSolution() `method

**Comment by sophialittlejohn** [comment]
The ticket has gone back into the hands of Jeannice and Dennis to figure out the necessary math for the multi tranche solver. 

TL;DR context:
We got stuck trying to implement the solver for two reasons:
1. the excel sheet used for the math solved complex cases nicely but couldn't handle the simple cases
2. the implementation was based on the excel sheet which used a method other than LP (linear programming) to solve the optimization problem. This proved to limit the implementation because we can't use fractions with a variable in the denominator in LP (e.g `1/trancheValue` would not be permitted for LP solution, but would have been necessary with the excel solver formula). We are limited to using LP for the implementation because we've had the library specifically built to handle big numbers and we don't have an alternative.

**Comment by sonam-jo** [comment]
this is for multi-tranche. no need for it atm, also we will have admin only solutions soon, making this unnessecary
